h2o-core/src/test/java/water/fvec/persist/FramePersistTest.java

```diff
@@ -0,0 +1,113 @@
+package water.fvec.persist;
+
+import hex.CreateFrame;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import water.DKV;
+import water.Key;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.runner.CloudSize;
+import water.runner.H2ORunner;
+import water.util.TwoDimTable;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static water.TestUtil.*;
+
+@RunWith(H2ORunner.class)
+@CloudSize(1)
+public class FramePersistTest {
+
+    @Rule
+    public TemporaryFolder temp = new TemporaryFolder();
+
+    @Test
+    public void testSaveAndLoadDouble() throws IOException {
+        Scope.enter();
+        try {
+            Vec v = Scope.track(createRandomDoubleVec(20, 42));
+            DKV.put(v);
+            Frame f = Scope.track(new Frame(Key.make(), new Vec[] { v }));
+            DKV.put(f);
+            TwoDimTable origData = f.toTwoDimTable(0, (int) v.length(), false);
+            File dest = temp.newFolder();
+            new FramePersist(f).saveTo(dest.getAbsolutePath(), false).get();
+            f.remove(true);
+            Frame returned = Scope.track(FramePersist.loadFrom(f._key, dest.getAbsolutePath()).get());
+            Frame loaded = DKV.get(f._key).get();
+            assertEquals(returned._key, loaded._key);
+            TwoDimTable loadedData = loaded.toTwoDimTable(0, (int) loaded.numRows(), false);
+            assertTwoDimTableEquals(origData, loadedData);
+        } finally {
+            Scope.exit();
+        }
+    }
+
+    @Test
+    public void testSaveAndLoadCategorical() throws IOException {
+        Scope.enter();
+        try {
+            Vec v = Scope.track(createRandomCategoricalVec(10_000, 42));
+            DKV.put(v);
+            Frame f = Scope.track(new Frame(Key.make(), new Vec[] { v }));
+            DKV.put(f);
+            TwoDimTable origData = f.toTwoDimTable(0, (int) v.length(), false);
+            File dest = temp.newFolder();
+            new FramePersist(f).saveTo(dest.getAbsolutePath(), false).get();
+            f.remove(true);
+            Scope.track(FramePersist.loadFrom(f._key, dest.getAbsolutePath()).get());
+            Frame loaded = DKV.get(f._key).get();
+            TwoDimTable loadedData = loaded.toTwoDimTable(0, (int) loaded.numRows(), false);
+            assertTwoDimTableEquals(origData, loadedData);
```

Comment:
why is the comparison done on 2d table? can we use assertFrameEquals instead? you just need to deepclone the original Frame

Reply:
I just wanted to have a purely local, cluster independent representation of the data. making a deep clone of the frame sounds reasonable too